cmd/server/app/webhook_update.go

```diff
@@ -0,0 +1,200 @@
+//
+// Copyright 2023 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/stacklok/minder/internal/config"
+	serverconfig "github.com/stacklok/minder/internal/config/server"
+	"github.com/stacklok/minder/internal/crypto"
+	"github.com/stacklok/minder/internal/db"
+	"github.com/stacklok/minder/internal/logger"
+	"github.com/stacklok/minder/internal/providers"
+	ghprovider "github.com/stacklok/minder/internal/providers/github"
+	provifv1 "github.com/stacklok/minder/pkg/providers/v1"
+)
+
+func cmdWebhookUpdate() *cobra.Command {
+	var updateCmd = &cobra.Command{
+		Use:   "update",
+		Short: "update the webhook configuration",
+		Long:  `Command to upgrade webhook configuration`,
+		RunE:  runCmdWebhookUpdate,
+	}
+
+	updateCmd.Flags().StringP("provider", "p", "github", "what provider type to update")
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+
+	if err := updateCmd.MarkFlagRequired("provider"); err != nil {
+		fmt.Fprintf(os.Stderr, "Error marking flag as required: %s\n", err)
+		os.Exit(1)
+	}
+
+	return updateCmd
+}
+
+func runCmdWebhookUpdate(cmd *cobra.Command, _ []string) error {
+	cfg, err := config.ReadConfigFromViper[serverconfig.Config](viper.GetViper())
+	if err != nil {
+		return fmt.Errorf("unable to read config: %w", err)
+	}
+
+	ctx := logger.FromFlags(cfg.LoggingConfig).WithContext(context.Background())
+
+	providerName := cmd.Flag("provider").Value.String()
+
+	cryptoEng, err := crypto.EngineFromAuthConfig(&cfg.Auth)
+	if err != nil {
+		return fmt.Errorf("failed to create crypto engine: %w", err)
+	}
+
+	dbConn, _, err := cfg.Database.GetDBConnection(ctx)
+	if err != nil {
+		return fmt.Errorf("unable to connect to database: %w", err)
+	}
+	defer func(dbConn *sql.DB) {
+		err := dbConn.Close()
+		if err != nil {
+			zerolog.Ctx(ctx).Error().Err(err).Msg("error closing database connection")
+		}
+	}(dbConn)
+
+	store := db.NewStore(dbConn)
+	allProviders, err := store.GlobalListProvidersByName(ctx, providerName)
```

Comment:
Should we instead iterate all providers, check that they implement github, and then attempt the migration? We're moving away from predetermined names and more into traits and classes

Reply:
We can do that, sure.

Reply:
done!